### PR TITLE
Save/restore 3D map views on project write/read

### PR DESCRIPTION
**Pull request by wonder-sk**
This makes 3D map view dock widgets behave like 2D map canvas dock widgets.

### CODE REVIEW

**Comment by NathanW2** [comment]
Nice!

File: src/app/qgisapp.cpp [hunk 1]
```diff
@@ -9955,26 +9975,68 @@ void QgisApp::new3DMapCanvas()
...
-  QgsFlatTerrainGenerator *flatTerrain = new QgsFlatTerrainGenerator;
-  flatTerrain->setCrs( map->crs() );
-  flatTerrain->setExtent( fullExtent );
-  map->setTerrainGenerator( flatTerrain );
+  bool existing = true;
+  const QList< Qgs3DMapCanvas * > existingCanvases = findChildren< Qgs3DMapCanvas * >();
+  QString name;
+  while ( existing )
+  {
+    name = tr( "3D Map %1" ).arg( i++ );
```
Comment (nyalldawson) [inline src/app/qgisapp.cpp]: Random thought ... But is "3d map" the right term here? Should it be "3d view" instead? "3d scene"?

Comment (wonder-sk) [reply src/app/qgisapp.cpp]: I do not really have any preference here - we can use whatever sounds most natural to native speakers.

Hopefully my favorite "Ultimate 3D Map View Pro (TM)" can be considered as well when choosing the right name.

Comment (nirvn) [reply src/app/qgisapp.cpp]: @nyalldawson , @wonder-sk , while I have no strong feeling here, I do like 3D Map [View], as it sits quite nicely next to the Map [View]:
screenshot from 2017-10-19 10-09-57

In any case, let's merge this 😉

**Pull request merged**